example-app/src/app/local-state/local-state.component.ts

```diff
@@ -0,0 +1,50 @@
+import { ChangeDetectionStrategy, Component, ChangeDetectorRef, OnDestroy, OnInit } from '@angular/core';
+import { ActionsSubject, Action } from '@ngrx/store';
+
+import { reducer, LocalState, GetManufacturersAction, INITIAL_LOCAL_STATE, FORM_ID } from './local-state.reducer';
+import { SetValueAction } from 'ngrx-forms';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'ngf-local-state',
+  templateUrl: './local-state.component.html',
+  styleUrls: ['./local-state.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LocalStateComponent implements OnInit, OnDestroy {
+
+  public localState: LocalState = INITIAL_LOCAL_STATE;
+
+  private subscription: Subscription;
+
+  constructor(private actionsSubject: ActionsSubject, private cd: ChangeDetectorRef) {
+  }
+
+  ngOnInit() {
+    this.subscription = this.actionsSubject.subscribe(action => {
+      const updated = this.handleAction(action);
+      if (updated) {
+        // since OnPush is used, need to trigger detectChanges
+        // when action from outside updates localState
+        this.cd.detectChanges();
+      }
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
```

Comment:
a tiny detail I like to keep in mind here is that `ngOnDestroy` can be called without `ngOnInit`, and therefore `this.subscription` can be undefined. You can either surround this with an `if` or initialize the variable upon declaration.